README.md

```diff
@@ -17,240 +20,154 @@ of the [Privacy Community Group](https://privacycg.github.io/).
+> TODO: figure out if we can put these behind user activation.
 
-```
-navigator.recordLogout(optionalUsername) –> Promise<void>
-```
+#### JS API
 
-The optional username parameter highlights that we might want to support
-concurrent logins on the same website which would require the site to
-keep track of who to log out and credential tokens to be scoped to user
-names.
+Here’s how the API for recording a login / logout could look:
 
-Here’s how the API for checking the login status could look:
+```javascript
+partial interface Navigator {
+  Promise<void> setLoggedIn(optional LoginStatusOptions options);
+  Promise<void> setLoggedOut(optional LogoutStatusOptions options);
+};
 
+interface LoginStatusOptions {
+  // To be extended by specific web platform features
+}
+
+interface LogoutStatusOptions {
+  // To be extended by specific web platform features
+}
 ```
-navigator.checkLoginState() –> Promise<bool>
-```
 
-This last API could potentially be allowed to be called by third-party
-iframes that do not currently have access to their cookies and website
-data. The iframes may want to render differently depending on whether
-the user is one of their logged in customers or not.
-
-### Defending Against Abuse
-
-If websites were allowed to set login status whenever they want, it
-would not constitute a trustworthy signal and would most likely be 
-abused for user tracking. We must therefore make sure that login status
-can only be set when the browser is convinced that the user meant to log
-in or the user is already logged in and wants to stay logged in.
-
-Another potential for abuse is if websites don’t call the logout API
-when they should. This could allow them to maintain the privileges tied
-to login status even after the user logged out.
-
-There are several ways the browser could make sure the login status
-is trustworthy:
-
-* Require websites to use WebAuthn or a password manager (including
-  Credential Management) before calling the API.
-* Require websites to take the user through a login flow according to
-  rules that the browser can check. This would be the escape hatch for
-  websites who can’t or don’t want to use WebAuthn or a password manager
-  but still want to set login status.
-* Show browser UI acquiring user intent when login status is set.
-  Example: A prompt.
-* Continuously show browser UI indicating an active login session on
-  the particular website. Example: Some kind of indicator in the URL
-  bar.
-* Delayed browser UI acquiring user intent to stay logged in, shown some
-  time after the login status was set. Example: Seven days after login
-  status was set – “Do you want to stay logged in to news.example?”
-* Requiring engagement to maintain login status. Example: Require user
-  interaction as first party website at least every N days to stay 
-  logged in. The browser can hide instead of delete the credential token
-  past this kind of expiry to allow for quick resurrection of the login
-  session.
```

Comment:
What is the justification for the removal of this section and replacement with the much weaker “Assumption of Abuse”? It seems like the new explainer doesn’t require or even permit UAs to try to prevent dishonest use of the login status API, and instead tells them to assume the signal may be dishonest. This defeats the original purpose of this API as described by @johnwilander .

Reply:
I think spirit of this section is represented through the explainer via specific and concrete ways in which a browser can "defending against abuse". That is, rather than a section on "ways the browser could make sure the login status is trustworthy" we are proposing "two specific ways (FedCM and SSA) that the browser could read this signal and trust its veracity -- and, in the future, if we find other ways, we can extend this API to introduce them".

I'd be happy to bring parts of this section back, but I hope that we didn't go as far as defeating the original purpose of this API as @johnwilander intended it to be.

If we did, and you and @johnwilander  don't feel that this explainer reconciles with the spirit Login Status API, we could still go back to using our FedCM-specific Sign-in-Status API, and we'd be OK with that too.

Reply:
> I'd be happy to bring parts of this section back, but I hope that we didn't go as far as defeating the original purpose of this API as @johnwilander intended it to be.

Maybe we could start by iterating on the PR to bring all of it back? I do not understand why deleting any of it is necessary to your other changes. If it is, please explain.

> If we did, and you and @johnwilander don't feel that this explainer reconciles with the spirit Login Status API, we could still go back to using our FedCM-specific Sign-in-Status API, and we'd be OK with that too.

I'm confused. Are you saying you _are_ willing to bring back some/all of this section (b/c I think removing it does not align with the original spirit of Login Status API), or we have to take this PR as-is or you'll diverge and ship your own thing?

Reply:
I do think that there's an initial conflict between the goals stated in the original explainer ("If websites were allowed to set login status whenever they want, it would not constitute a trustworthy signal") and how this PR changes things (by allowing websites to set login status whenever they want).

As Sam mentions in https://github.com/privacycg/is-logged-in/pull/54#issuecomment-1691974190 this can probably be overcome, so I think the larger question here is can we find some convergence between the two ideas that leaves the door open for WebKit's login status, for the benefit of developers who wouldn't have to deal with two very similar signals.

If we think we can, then I don't think it makes sense to restore this section as-is, as that would be incompatible with some new ideas presented in this PR. Rather, we should adapt it to whatever compromise we arrive at (e.g. a `prompt` or `trusted` flag in the `setLoggedIn` call as Sam suggests in his comment).

Reply:
> We could do those things, but shouldn't trustworthiness considerations in the Explainer cover the whole span of envisioned use cases, not just the FedCM one?

I think the key part here to make sure we are on the same page is to whether to capture in the explainer (and the spec) what we **hypothetically could do** or what we are **actively planning to do**: we propose we focus on the latter. We don't have any active plans to use this API outside of the direct use of FedCM and (possibly) SAA, which are both described in the explainer as concrete things that we would like to do do. We think it is **plausible** that it could hypothetically be used for other things, so we think it makes sense to have this more general API, but we wanted to be clear that we haven't yet found other uses where the design of incentives and the economics of abuse align.

> I'm confused. Are you saying you are willing to bring back some/all of this section (b/c I think removing it does not align with the original spirit of Login Status API), or we have to take this PR as-is or you'll diverge and ship your own thing?

It is important to make sure that we are clear that we haven't found (nor are proposing) use of the Login Status API **outside** of FedCM (and, as I said, possibly, the Storage Access API), so if you are expecting that this proposal will offer a concrete way in which a website can declare their Login Status in a way that we think that the browser can generally trust (again, outside of how FedCM/SAA would use), then yes, I think this proposal isn't aligned with the original spirit of the Login Status API (and hence, naturally, we'd be happy to make it FedCM-specific).

We do, however, remain of the opinion that, while this is a **constrained** use of the Login Status API (for FedCM and SAA specifically), it can be designed to be **extensible** for further user, and hence, is worth using a generalizable/extensible API, so our preference is this PR to be accepted so that we can make FedCM depend on it.

Reply:
I don’t know where it got lost in this conversation, but we’ve been clear from the get go that we want a trustworthy signal as to 1) which websites it’s most important to maintain state for, and 2) which websites the user has an established relationship with and is more likely to want to enable powerful but risky features for.

Reply:
> I don’t know where it got lost in this conversation, but we’ve been clear from the get go that we want a trustworthy signal as to 1) which websites it’s most important to maintain state for, and 2) which websites the user has an established relationship with and is more likely to want to enable powerful but risky features for.

I think the disconnect may be on how we are using the term `trustworthy`: what's written in this PR **is** trustworthy within the thread model of the specific proposals trying to consume it (FedCM and SAA).

We haven't thought it through, but the intuition is that it is not `trustworthy` within other threat models, including ones involving what's listed here (e.g. clearing storage, etc), so we are hesitant in introducing those modals without having any plans to ship them.

In this proposal, we are (a) proposing a couple of uses that we think that the bit can be used in a `trustworthy` way (again, within the threat model of how FedCM and SAA operate) and (b) leaving the extensibility points so that other abuse models can be introduced as it goes along.